src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -996,83 +999,78 @@ class npc_dkc1_gothik : public CreatureScript
 
 };
 
-class npc_scarlet_ghoul : public CreatureScript
+struct npc_scarlet_ghoul : public ScriptedAI
 {
-public:
-    npc_scarlet_ghoul() : CreatureScript("npc_scarlet_ghoul") { }
+    npc_scarlet_ghoul(Creature* creature) : ScriptedAI(creature)
+    {
+        Initialize();
+    }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void Initialize()
     {
-        return new npc_scarlet_ghoulAI(creature);
+        if (urand(0, 1))
+            if (Unit* owner = me->GetOwner())
+                Talk(SAY_LINE_0, owner);
```

Comment:
Initialize() is supposed to only set variables to their default values, so it can be called in the constructor and in Reset(). Please move Talk() calls out of the constructor

Reply:
You mean constructor is called also on Reset()?

Reply:
No, he means Initialize is not supsoed to be used for anything that's not used for initialization. (members, variables etc.)

Use JustAppeared() override instead.

Reply:
This bugs creature (not following owner).

Reply:
call ScriptedAI::JustEngaged() in the hook then. This will execute the default follow just appeared function which will trigger the follow movement.

Reply:
> This bugs creature (not following owner).

did you call the base JustAppeared() too ?

Reply:
Thanks, didn't think about that. Works now 💃

Reply:
> Thanks, didn't think about that. Works now 💃

the feeling of learning 👍 😃